File: C#/acronym/Acronym.cs
using System;

public static class Acronym
{
    // variables
    static string result;
    // abbreviates a phrase
    public static string Abbreviate(string phrase)
    {
        result = "";
        string[] words = phrase.Split(" ");

        // if word has - or _, proceed to special processing. otherwise, add first letter to result
        foreach (string word in words)
        {
            if (word.IndexOf('-') >= 0 || word.IndexOf('_') >= 0)
            {
                CheckSpecialCharacters(word);
            }
            else
            {
                result += Char.ToUpper(word[0]);
            }
        }

        return result;
    }

    // checks words with - and _ to see if need to add to result
    public static void CheckSpecialCharacters(string word)
    {
        bool hasDash = word.IndexOf('-') >= 0;
        bool hasUnderscore = word.IndexOf('_') >= 0;

        if (hasDash && word.Length > 1)
        {
            string[] moreWords = word.Split('-');
            if (moreWords.Length > 0)
            {
                result += Char.ToUpper(moreWords[0][0]);   // first element, first position
                result += Char.ToUpper(moreWords[1][0]);   // second element, first position
            }
        }

        if (hasUnderscore && word.Length > 1)
        {
            string[] moreWords = word.Split('_', StringSplitOptions.RemoveEmptyEntries);
            if (moreWords.Length > 0)
            {
                result += Char.ToUpper(moreWords[0][0]);   
            }
        }
    }
}

File: JavaScript/acronym.js
export const parse = (phrase) => {
  let result = '';
  let words = phrase.split(' ');

  // if word has - or _, proceed to special processing. otherwise, add first letter to result
  words.forEach(word => {
    if (word.indexOf('-') >= 0 || word.indexOf('_') >= 0) {
      checkSpecialCharacters(word);
    }
    else {
      result += word[0].toUpperCase();
    }
  })

  // checks words with - and _ to see if need to add to result
  function checkSpecialCharacters(word) {
    let hasDash = word.indexOf('-') >= 0;
    let hasUnderscore = word.indexOf('_') >= 0;
    let moreWords = [];

    if (hasDash && word.length > 1) {
      moreWords = word.split('-').filter(element => element.length > 0);
      result += moreWords[0][0].toUpperCase();   // first element, first position
      result += moreWords[1][0].toUpperCase();   // second element, first position
    }

    if (hasUnderscore && word.length > 1) {
      moreWords = word.split('_').filter(element => element.length > 0);
      result += moreWords[0][0].toUpperCase();
    }
  }

  return result;
};

